Python - Update Contact using Python Nylas SDK

SouvikDcoder opened the pull request:
## PR Acceptance Criteria

## Guidelines
* [x] PR follows the [Contributing guidelines](https://github.com/nylas-samples/nylas-hacktoberfest-2022/blob/main/CONTRIBUTION.md)?
* [x] PR Adheres to our [Code of Conduct](https://github.com/nylas-samples/nylas-hacktoberfest-2022/blob/main/CODE_OF_CONDUCT.md)?
* [x] Have you checked to ensure there aren't other open Pull Requests for the same update/change?
* [x] Ensure you are assigned the issue before creating a PR.
* [x] Only one issue per PR is allowed. This simplifies the code review.

## Code complete
* [x] A `README.md` available describing 1) how to setup, 2) how to test the code and 3) the expected output? Here is an [example](code-samples/account/node/read-account/readme.md). 
* [x] The code sample in the correct location (i.e. `code-samples/contact/python/update_contact/`)?
* [x] Link added to the repository readme (view `# Code Snippets created by the Nylas Community` with examples) describing what the code sample does for others to try out
* [x] The code can be run using Github Codespaces, so the maintainer can try out the code sample remotely to ensure everything is working before we accept the PR.

## For Maintainer Use Only (does not apply to PR Author)
* [ ] Purpose of code sample is clear
* [ ] The code sample is readable
* [ ] Steps try out the code are provided
* [ ] Code sample runs on codespaces environment
* [ ] Code works as expected
* [ ] PR label `hacktoberfest-accepted` added

SouvikDcoder commented:
This PR contains the changes required for issues: https://github.com/nylas-samples/nylas-hacktoberfest-2022/issues/13
Kindly review and merge. Do let me know if any changes are required.
Thank You!

relaxedtomato commented on code-samples/contact/python/update_contact/update_contact.py:
```diff
@@ -0,0 +1,73 @@
+# Set environment variables
+os.environ['CLIENT_ID'] = "The ID for your application"
+os.environ['CLIENT_SECRET'] = "The secret for your application"
+os.environ['ACCESS_TOKEN'] = "The access token you generated with the Quick Start application."
```
can we follow [this approach](https://github.com/nylas-samples/nylas-hacktoberfest-2022/blob/main/code-samples/contact/python/read_contacts/read_contacts.py#L5) for setting environment variables?

I see we've done this in the past, but having your environment variables stored in an `.env` file that is part of `.gitignore` is a better practice.

SouvikDcoder commented:
@relaxedtomato Review comments are accommodated and workflow is updated. Kindly review.

relaxedtomato commented:
@SouvikDcoder - I received the following error, did you receive the same:
```Reason: d5tgr4flai636ool27g5cxagx is an auto-generated contact and this operation isnt supported for auto-generated contacts.. Nylas Error Type: api_error```

So it seems we cannot update a newly created contact - can you update the code as such:
- Include `CONTACT_ID` as part of the environment variable
- Update readme [with a link to the code to read contacts](https://github.com/nylas-samples/nylas-hacktoberfest-2022/tree/main/code-samples/contact/python/read_contacts) for the user to pick a CONTACT_ID
- I recommend remove all code related to creating a contact, since that's [already found here](https://github.com/nylas-samples/nylas-hacktoberfest-2022/tree/main/code-samples/contact/python/create_contact)

relaxedtomato commented on code-samples/contact/python/update_contact/update_contact.py:
```diff
@@ -0,0 +1,69 @@
+contact.im_addresses['gtalk'] = 'Google'
+
+# Save the contact to Nylas with the updated configurtations
+contact.save()
```
I recommend remove all code related to creating a contact, since that's [already found here](https://github.com/nylas-samples/nylas-hacktoberfest-2022/tree/main/code-samples/contact/python/create_contact)

SouvikDcoder replied:
Code removed.

relaxedtomato commented on code-samples/contact/python/update_contact/update_contact.py:
```diff
@@ -0,0 +1,69 @@
+nylas = APIClient(
+    os.environ.get("CLIENT_ID"),
+    os.environ.get("CLIENT_SECRET"),
+    os.environ.get("ACCESS_TOKEN"),
```
Let's just add an option to retrieve CONTACT_ID, so `os.environ.get("CONTACT_ID"),`

relaxedtomato commented on code-samples/contact/python/update_contact/readme.md:
```diff
@@ -0,0 +1,128 @@
+## Usage
+
+Execute the python files using command line or in any editor of your choice ensuring python is setup. Update below values in your `.env` file. 
+
```
For usage, include the error if you use an auto-generated contact, recommend the user to use a contact not created through Nylas.
```
nylas.client.errors.NylasApiError: 400 Bad Request. Reason: d5tgr4flai636ool27g5cxagx is an auto-generated contact and this operation isnt supported for auto-generated contacts.. Nylas Error Type: api_error
```

SouvikDcoder replied:
`readme.md` file is updated with the added note.

relaxedtomato reviewed (changes requested):
view comments, we need to clean up the code and address the error I saw

SouvikDcoder commented:
@relaxedtomato Updated the PR with the requested changes. Kindly review. 
Let me know if any other changes are required. 
Changes accommodated : 
- Updated readme file with the latest workflow changes 
- Added contact ID to `.env` file 
- Removed the create contact code from the update contact python file

relaxedtomato commented on code-samples/contact/python/update_contact/readme.md:
````diff
@@ -0,0 +1,102 @@
+```
+## Pre-requisite 
+
+Please add the customer id from a contact that was created using Nylas lib in the `.env` file in the below format.
````
```suggestion
Please add the contact id from an existing contact in the `.env` file in the below format.
```

relaxedtomato commented on code-samples/contact/python/update_contact/readme.md:
```diff
@@ -0,0 +1,102 @@
+
+## Note : Expected Errors
+
+If you try to update a contact that was not created using a Nylas lib, please expect below error traceback.
```
```suggestion
If you try to update a contact that was auto-created using Nylas, you will receive the below error.
```

relaxedtomato commented on code-samples/contact/python/update_contact/readme.md:
````diff
@@ -0,0 +1,102 @@
+```
+CONTACT_ID = <contact id>
+```
+Read the contact details and fetch a contact ID from here `/code-samples/contact/python/read_contacts/read_contacts.py`  
````
```suggestion
Read the contact details and fetch a contact ID using the [read contacts code sample](https://github.com/nylas-samples/nylas-hacktoberfest-2022/tree/main/code-samples/contact/python/read_contacts)
```

relaxedtomato reviewed (changes requested):
minor changes

relaxedtomato commented:
@SouvikDcoder this one is close to ready, let me know if you can address the comments.

SouvikDcoder commented:
@relaxedtomato comments addressed. Kindly review.

SouvikDcoder commented:
Thanks, @relaxedtomato @atejada 😄
